File: python/interview_22.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getKthFromEnd(self, head: ListNode, k: int) -> ListNode:
        prev = head
        for i in range(k):
            head = head.next
        while head is not None:
            head = head.next
            prev = prev.next
        return prev

File: Java/interview_22.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode getKthFromEnd(ListNode head, int k) {
        ListNode prev = head;
        for (int i = 0; i < k; ++i) {
            head = head.next;
        }
        while (head != null) {
            head = head.next;
            prev = prev.next;
        }
        return prev;
    }
}
